misc/incremental_checker.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+"""
+This file compares the output and runtime of running normal vs
+incremental mode on the history of any arbitrary git repo as a
+way of performing a sanity check to make sure incremental mode
+is working correctly and efficiently.
+
+This script will, by default, run all tests against mypy's repo.
+"""
+
+from typing import Any, Dict, List, Tuple
+
+from argparse import ArgumentParser, RawDescriptionHelpFormatter, ArgumentDefaultsHelpFormatter
+import json
+import os
+import shutil
+import subprocess
+import sys
+import textwrap
+import time
+
+
+LOOK_BACK = 30
+CACHE_PATH = ".incremental_checker_cache.json"
+MYPY_REPO_URL = "https://github.com/python/mypy.git"
+MYPY_MODULE_NAME = "mypy"
+
+JsonDict = Dict[str, Any]
+
+
+def print_offset(text: str, indent_length: int = 4) -> None:
+    print()
+    print(textwrap.indent(text, ' ' * indent_length))
+    print()
+
+
+def delete_folder(folder_path: str) -> None:
+    if os.path.exists(folder_path):
+        shutil.rmtree(folder_path)
+
+
+def execute(command: List[str], fail_on_error: bool = True) -> Tuple[str, str, int]:
+    proc = subprocess.Popen(
+        ' '.join(command),
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        shell=True)
+    stdout_bytes, stderr_bytes = proc.communicate()  # type: Tuple[bytes, bytes]
+    stdout, stderr = stdout_bytes.decode('utf-8'), stderr_bytes.decode('utf-8')
+    if fail_on_error and proc.returncode != 0:
+        print("RETURN CODE:", proc.returncode)
+        print()
+        print('STDOUT:')
+        print_offset(stdout)
+        print('STDERR:')
+        print_offset(stderr)
+        raise Exception("Unexpected error from external tool.")
```

Comment:
Can you raise a more specific exception here? You don't have to create a new exception class -- just using RuntimeError would be better than Exception already.